__tests__/__unit__/dataset/actions.unit.test.ts

```diff
@@ -1557,3 +1561,383 @@ describe("Dataset Actions Unit Tests - Function hRecallDataSet", () => {
+        node.contextValue = globals.DS_SESSION_CONTEXT;
+        const getChildrenSpy = jest.spyOn(node, "getChildren");
+        getChildrenSpy.mockResolvedValue([]);
+
+        mocked(vscode.window.showQuickPick).mockResolvedValueOnce("Data Set Binary" as any);
+        await dsActions.createFile(node, blockMocks.testDatasetTree);
+        expect(mocked(vscode.workspace.getConfiguration)).lastCalledWith("Zowe-Default-Datasets-Binary");
+
+        mocked(vscode.window.showQuickPick).mockResolvedValueOnce("Data Set C" as any);
+        await dsActions.createFile(node, blockMocks.testDatasetTree);
+        expect(mocked(vscode.workspace.getConfiguration)).lastCalledWith("Zowe-Default-Datasets-C");
+
+        mocked(vscode.window.showQuickPick).mockResolvedValueOnce("Data Set Classic" as any);
+        await dsActions.createFile(node, blockMocks.testDatasetTree);
+        expect(mocked(vscode.workspace.getConfiguration)).lastCalledWith("Zowe-Default-Datasets-Classic");
+
+        mocked(vscode.window.showQuickPick).mockResolvedValueOnce("Data Set Partitioned" as any);
+        await dsActions.createFile(node, blockMocks.testDatasetTree);
+        expect(mocked(vscode.workspace.getConfiguration)).lastCalledWith("Zowe-Default-Datasets-PDS");
+
+        mocked(vscode.window.showQuickPick).mockResolvedValueOnce("Data Set Sequential" as any);
+        await dsActions.createFile(node, blockMocks.testDatasetTree);
+        expect(mocked(vscode.workspace.getConfiguration)).lastCalledWith("Zowe-Default-Datasets-PS");
+
+        // tslint:disable-next-line:no-magic-numbers
+        expect(createDataSetSpy).toHaveBeenCalledTimes(5);
+    });
+    it("Checking of proper configuration being picked up for different DS types with credentials prompt for favorite", async () => {
+        createGlobalMocks();
+        const blockMocks = createBlockMocks();
+
+        blockMocks.profileInstance.promptCredentials.mockReturnValue(["fake", "fake", "fake"]);
+        blockMocks.testDatasetTree.createFilterString.mockResolvedValue("test");
+        blockMocks.testDatasetTree.getHistory.mockReturnValue([]);
+        mocked(Profiles.getInstance).mockReturnValue(blockMocks.profileInstance);
+        mocked(vscode.window.showInputBox).mockResolvedValue("test");
+        const createDataSetSpy = jest.spyOn(blockMocks.mvsApi, "createDataSet");
+        createDataSetSpy.mockReset();
+        const node = new ZoweDatasetNode("HLQ.TEST.TO.NODE", vscode.TreeItemCollapsibleState.None, null, blockMocks.sessionWithoutCredentials);
+        node.contextValue = globals.DS_SESSION_CONTEXT + globals.FAV_SUFFIX;
+        const getChildrenSpy = jest.spyOn(node, "getChildren");
+        getChildrenSpy.mockResolvedValue([]);
+
+        mocked(vscode.window.showQuickPick).mockResolvedValueOnce("Data Set Binary" as any);
+        await dsActions.createFile(node, blockMocks.testDatasetTree);
```

Comment:
1 call per it() block

Reply:
Answering on all at once, specific of this test is verification of proper run with different types without details, splitting it will result in hundreds of lines of the duplicates. I think that's an exception which can be made, if no - fine, I will spend time splitting it to 5 test cases with duplicates.

Reply:
Yeah, I guess that's fine. I'm just being OCD about it...